airbyte-integrations/connectors/source-mysql/src/main/resources/spec.json

```diff
@@ -54,11 +54,140 @@
+                "type": "string",
+                "const": "disable",
+                "enum": ["disable"],
+                "default": "disable",
+                "order": 0
+              }
+            }
+          },
+          {
+            "title": "preferred",
+            "description": "Preferred SSL mode.",
+            "required": ["mode"],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "preferred",
+                "enum": ["preferred"],
+                "default": "preferred",
+                "order": 0
+              }
+            }
+          },
+          {
+            "title": "required",
+            "description": "Require SSL mode.",
+            "required": ["mode"],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "required",
+                "enum": ["required"],
+                "default": "required",
+                "order": 0
+              }
+            }
+          },
+          {
+            "title": "Verify CA",
+            "description": "Verify CA SSL mode.",
+            "required": ["mode"],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "verify_ca",
+                "enum": ["verify_ca"],
+                "default": "verify_ca",
+                "order": 0
+              },
+              "ca_certificate": {
+                "type": "string",
+                "title": "CA certificate",
+                "description": "CA certificate",
+                "airbyte_secret": true,
+                "multiline": true,
+                "order": 1
+              },
+              "client_key_password": {
+                "type": "string",
+                "title": "Client key password (Optional)",
+                "description": "Password for keystorage. This field is optional. If you do not add it - the password will be generated automatically.",
+                "airbyte_secret": true,
+                "order": 2
+              }
+            }
+          },
+          {
+            "title": "Verify Identity",
+            "description": "Verify-full SSL mode.",
+            "required": [
+              "mode",
+              "ca_certificate",
+              "client_certificate",
+              "client_key"
+            ],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "verify_identity",
+                "enum": ["verify_identity"],
+                "default": "verify_identity",
+                "order": 0
+              },
+              "ca_certificate": {
+                "type": "string",
+                "title": "CA certificate",
+                "description": "CA certificate",
+                "airbyte_secret": true,
+                "multiline": true,
+                "order": 1
+              },
+              "client_certificate": {
```

Comment:
as far as I understand, when connecting with mode `VERIFY CA`, client certificates are not used at all. As you can see, when connecting using this mode, I only use the `trustStore`, to which I add the root certificate, and the `keyStore`, is not used at all. At the same time, the connection works fine. (In case of using `VERIFY_IDENTITY` mode, `keyStore` validation is mandatory and the connection will not work without it)

Reply:
This is not very clear form docs, but VERIFY_CA and VERIFY_IDENTITY are both related to the way the client (being out connector) is verifying the server certificate.
VERIFY_CA: Any server certificate that is trusted will allow the connection to go through.
VERIFY_IDENTITY: In addition to trust, The name on the server certificate has to match the hostname to which client is connecting (i.e connecting to mysql host example.com, the name on the certificate has to be example.com).

Both modes allow the client to connect with or without a client certificate.
In command line you can do : `mysql -ussl2 -h 127.0.0.1 -P 2000 -ppassword --ssl-mode=VERIFY_CA --ssl-ca=ca.pem --ssl-cert=client-cert.pem --ssl-key=client-key.pem`

and also `mysql -ussl2 -h 127.0.0.1 -P 2000 -ppassword --ssl-mode=VERIFY_IDENTITY --ssl-ca=ca.pem`

Reply:
I understood you. I added the changes you wrote about. Client certificates can now be used for `VERIFY_CA` and `VERIFY_IDENTITY` connections. Tested it both manually and using test containers.